linera-base/src/data_types.rs

```diff
@@ -842,21 +863,77 @@ impl HashedBlob {
     }
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(rename = "HashedBlob")]
+struct SerializableHashedBlob {
+    pub id_type: String,
+    pub blob: Blob,
+}
+
 impl Serialize for HashedBlob {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
-        S: serde::Serializer,
+        S: Serializer,
     {
-        self.blob.serialize(serializer)
+        let id_type = match self.id {
+            BlobId::Data(_) => "User",
+        }
+        .to_string();
+
+        if serializer.is_human_readable() {
+            let bytes = bcs::to_bytes(&SerializableHashedBlob {
+                id_type,
+                blob: self.blob.clone(),
+            })
+            .map_err(serde::ser::Error::custom)?;
+            serializer.serialize_str(&hex::encode(bytes))
+        } else {
+            SerializableHashedBlob::serialize(
+                &SerializableHashedBlob {
+                    blob: self.blob.clone(),
+                    id_type,
+                },
+                serializer,
+            )
+        }
     }
 }
 
-impl<'a> Deserialize<'a> for HashedBlob {
+impl<'de> Deserialize<'de> for HashedBlob {
```

Comment:
I don't think it's super important (or desirable) to rename `'a` in `'de`. Please let's keep PR minimal.

Reply:
I rewrote it so didn't realize we were using `'a` before, can change it